torch/autograd/gradcheck.py

```diff
@@ -240,6 +324,25 @@ def fail_test(msg):
     return jacobians, failed
 
 
+def get_analytical_jacobian(inputs, output, nondet_tol=0.0, grad_out=1.0):
+    # Replicates the behavior of the old get_analytical_jacobian before the refactor
+    warnings.warn("get_analytical_jacobian is deprecated!")
+    diff_input_list = list(iter_tensors(inputs, True))
+
+    def backward_fn(grad_output):
+        return torch.autograd.grad(output, diff_input_list, grad_output,
+                                   retain_graph=True, allow_unused=True)
+
+    jacobians_rows = compute_analytical_jacobian_rows(backward_fn, output.clone(), grad_out)
+    jacobians_rows_reentrant = compute_analytical_jacobian_rows(backward_fn, output.clone(), grad_out)
+
+    jacobians, correct_grad_types, correct_grad_sizes = combine_jacobian_rows(jacobians_rows, inputs, output)
+    jacobians_reentrant, _, _ = combine_jacobian_rows(jacobians_rows_reentrant, inputs, output)
+    reentrant = check_jacobians_equal(jacobians, jacobians_reentrant, nondet_tol)
+
+    return jacobians, reentrant, correct_grad_sizes, correct_grad_types
```

Comment:
Since we're not using `get_analytical_jacobian` anymore in our testing code, could we add a correctness test for it to make sure that it still works? Ditto for `get_numerical_jacobian`.

It would also be nice to test that the deprecation message shows up properly.